### PR TITLE
Create a WAI lsp server

### LINKED ISSUE #47: Create a Language Server project in Rust
We should be able to reuse the [`tower-lsp`](https://docs.rs/tower-lsp/latest/tower_lsp/) crate to provide the server abstraction for our Language Server. From there, follow the `tower_lsp::LanguageServer` trait and see where it takes you.

The goal for this issue is to register the language server and provide a "Hello World" task so we know it works (e.g. something that pops up a message showing the version number). We **don't** need to implement anything WAI-specific, we just want to know that everything is wired up correctly.

We can copy a lot of the structure from [`IWANABETHATGUY/tower-lsp-boilerplate`](https://github.com/IWANABETHATGUY/tower-lsp-boilerplate).

### PR DESCRIPTION
**Pull request by dynamite-bud**
Closes #47.

### CODE REVIEW

File: wai-language-server/Cargo.toml [hunk 1]
```diff
@@ -0,0 +1,11 @@
+[package]
+name = "wai-language-server"
+version = "0.1.0"
+edition = "2021"
+
+# See more keys and their definitions at https://doc.rust-lang.org/cargo/reference/manifest.html
+
+[dependencies]
+serde_json = "1.0.91"
+tokio = { version = "1.25.0", features = ["full"] }
```
Comment (Michael-F-Bryan) [inline wai-language-server/Cargo.toml:10]: I think we should be able to trim the features we use and save a bit on compilation times.

```suggestion
tokio = { version = "1.25.0", features = ["rt", "rt-multi-thread", "macros"] }
```

File: wai-language-server/src/main.rs [hunk 2]
```diff
@@ -0,0 +1,78 @@
+use tower_lsp::jsonrpc::Error;
+use tower_lsp::jsonrpc::Result;
+use tower_lsp::lsp_types::MessageType;
+use tower_lsp::lsp_types::*;
+use tower_lsp::{Client, LanguageServer, LspService, Server};
```
Comment (Michael-F-Bryan) [inline wai-language-server/src/main.rs:5]: Can you merge these imports into a single line? i.e. `use tower_lsp::{jsonrpc::{Error, ...}, lsp_types::MessageType, ...}`.

We should also get rid of the glob import (`use tower_lsp::lsp_types::*`) and just import the things we need.

File: wai-language-server/src/main.rs [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+    client: Client,
+}
+
+#[tower_lsp::async_trait]
+impl LanguageServer for Backend {
+    async fn initialize(&self, _: InitializeParams) -> Result<InitializeResult> {
+        Ok(InitializeResult {
+            capabilities: ServerCapabilities {
+                hover_provider: Some(HoverProviderCapability::Simple(true)),
+                completion_provider: Some(CompletionOptions::default()),
```
Comment (Michael-F-Bryan) [inline wai-language-server/src/main.rs:20]: Let's get rid of the hover and completion providers for now.

File: wai-language-server/src/main.rs [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+            CompletionItem::new_simple("Bye".to_string(), "More detail".to_string()),
+        ])))
+    }
+
+    async fn hover(&self, _: HoverParams) -> Result<Option<Hover>> {
+        Ok(Some(Hover {
+            contents: HoverContents::Scalar(MarkedString::String("You're hovering!".to_string())),
+            range: None,
+        }))
+    }
```
Comment (Michael-F-Bryan) [inline wai-language-server/src/main.rs:53]: Let's get rid of these now we know the extension is talking to our LSP.

**Review by Michael-F-Bryan — CHANGES_REQUESTED** [review]

File: wai-language-server/Cargo.toml (hunk 1 above)
Comment (dynamite-bud) [reply wai-language-server/Cargo.toml:10]: We also needed the "io-std" feature

**Review by dynamite-bud — COMMENTED** [review]
We needed another `io-std` for tokio as well plus we can also remove `activationEvents` from `package.json`

**Review by Michael-F-Bryan — APPROVED** [review]

**Pull request merged**